### PR TITLE
Use require() for templates

### PR DESCRIPTION
**Pull request by robertknight**
This PR changes the way UI components fetch their HTML templates to use the same `require()` mechanism that is used for importing JS imports, instead of relying on the template being added to `app.html` as a `<script>` tag with the HTML as the content. Using `require()` for .html files is done via the `stringify` Browserify transform which converts HTML into JS strings.

There are a number of general reasons for doing this, plus it also helps specifically with fixing some issues with collapsible excerpts [1]:

* It makes components more self-contained, as each pulls in its own templates
* ~~It moves my drinks cabinet a few inches closer to Berlin~~ It simplifies the app.html build a little
* It makes behavior of all components the same, because they will all resolve their templates synchronously, whereas there were some components previously using `template` (which is sync) and others using `templateUrl` (which is async).
* It is faster, because it avoids going through a bunch of machinery internally in Angular that is used for fetching templates from the network (internally `templateUrl` fetches work exactly the same as an `$http` request, except that it uses a local cache which gets pre-populated from the `<script>` tags on the page).

[1] Specifically, it makes template fetching synchronous, which means that the `<excerpt>` post-link function is called, it can rely on its content having been fully processed and thus it can measure the height of that content. Subsequently the height can change, but in response to specific known events (eg. images loading, the sidebar being resized) which we can watch for separately.

### CODE REVIEW

File: scripts/gulp/create-bundle.js [hunk 1]
```diff
@@ -117,6 +118,12 @@ module.exports = function createBundle(config, buildOpts) {
 
   var bundle = browserify([], bundleOpts);
 
+  bundle.transform(stringify, {
```
Comment (nickstenning) [inline scripts/gulp/create-bundle.js]: All our other browserify transforms are defined in `package.json`. Should this not go there, too?

Comment (robertknight) [inline scripts/gulp/create-bundle.js]: Given that this transform is applied unconditionally, it probably should. I'll give it a go.

Comment (robertknight) [inline scripts/gulp/create-bundle.js]: Done.

**Comment by nickstenning** [comment]
LGTM.

**Pull request merged**